File: back-end/src/main/java/org/neuverse/entity/Usuario.java
package org.neuverse.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "usuarios", schema = "core")
public class Usuario extends PanacheEntityBase {

    @Id
    @Column(name = "id", nullable = false)
    public UUID id;

    @Column(name = "nome")
    public String nome;

    @Column(name = "email", nullable = false)
    public String email;

    @Column(name = "online", nullable = false)
    public Boolean online;

    @Column(name = "cargo")
    public String cargo;

    @Column(name = "senha_hash")
    public String senhaHash;

    @Column(name = "loja_id")
    public UUID lojaId;

    @Column(name = "criado_em")
    public OffsetDateTime criadoEm;

    @Column(name = "atualizado_em")
    public OffsetDateTime atualizadoEm;

    // ---- Novos campos ----
    @Column(name = "email_verificado")
    public Boolean emailVerificado;

    @Column(name = "two_factor_enabled")
    public Boolean twoFactorEnabled;

    @Column(name = "two_factor_secret")
    public String twoFactorSecret; // base32

    @Column(name = "ultimo_login_em")
    public OffsetDateTime ultimoLoginEm;

    public static Usuario findByEmail(String email) {
        return find("email", email).firstResult();
    }

    @PrePersist
    void prePersist() {
        if (id == null) id = UUID.randomUUID();
        if (online == null) online = Boolean.FALSE;
        if (emailVerificado == null) emailVerificado = Boolean.FALSE;
        if (twoFactorEnabled == null) twoFactorEnabled = Boolean.FALSE;
        if (criadoEm == null) criadoEm = OffsetDateTime.now();
        atualizadoEm = criadoEm;
    }

    @PreUpdate
    void preUpdate() { atualizadoEm = OffsetDateTime.now(); }
}

File: back-end/src/main/kotlin/org/neuverse/entity/Usuario.kt
package org.neuverse.entity

import jakarta.persistence.*
import java.time.OffsetDateTime
import java.time.ZoneOffset
import java.util.UUID

@Entity
@Table(name = "usuarios", schema = "core")
class Usuario(

    @Id
    @Column(name = "id", nullable = false, updatable = false)
    var id: UUID = UUID.randomUUID(),

    @Column(name = "nome", nullable = false)
    var nome: String = "",

    @Column(name = "email", nullable = false, unique = true)
    var email: String = "",

    @Column(name = "online", nullable = false)
    var online: Boolean = false,

    // 👇 agora mapeado como INT (Postgres int2 → Int em Kotlin)
    @Column(name = "cargo", nullable = false)
    var cargo: Int = 3,    // 1=ADMIN, 2=DIRETOR, 3=FUNCIONARIO

    @Column(name = "criado_em", nullable = false)
    var criadoEm: OffsetDateTime = OffsetDateTime.now(ZoneOffset.UTC),

    @Column(name = "atualizado_em", nullable = false)
    var atualizadoEm: OffsetDateTime = OffsetDateTime.now(ZoneOffset.UTC),

    @Column(name = "senha_hash", nullable = false)
    var senhaHash: String = "",

    @Column(name = "loja_id")
    var lojaId: UUID? = null,

    @Column(name = "email_verificado", nullable = false)
    var emailVerificado: Boolean = false,

    @Column(name = "two_factor_enabled", nullable = false)
    var twoFactorEnabled: Boolean = false,

    @Column(name = "two_factor_secret")
    var twoFactorSecret: String? = null,

    @Column(name = "ultimo_login_em")
    var ultimoLoginEm: OffsetDateTime? = null
)
